hr_holidays_natural_period/models/hr_leave.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class HrLeave(models.Model):
+    _inherit = "hr.leave"
+
+    def _get_number_of_days(self, date_from, date_to, employee_id):
+        if (
+            self.holiday_status_id
+            and self.holiday_status_id.request_unit == "natural_day"
+            and date_from
+            and date_to
+        ):
```

Comment:
Why do you need such complex condition?
```suggestion
        if self.holiday_status_id.request_unit == "natural_day":
```